docs/incidents/2021-06-29--actions-budget-terraform-state.md

```diff
@@ -0,0 +1,138 @@
+
+However, the second issue with Terraform was a much bigger problem. It turns out that @jaronoff97 used the `terraform import` command while working on [adding our bastion server’s security group to the Terraform configuration files (#274)][issue-274]. Unfortunately, `terraform import` modifies the *state* that Terraform uses to plan changes against. Since the corresponding config files were still on @jaronoff97’s pull request branch, that meant the Terraform config files on `main` and all other branches were out of sync with with our Terraform *state*, causing Terraform to plan to destroy important resources in AWS whenever trying to deploy from any branch except @jaronoff97’s pull request.
+
+Terraform Cloud keeps track of past versions of the state, but does not make it possible to roll back or revert to a particular state version. However, @Mr0grog was able to use Terraform Cloud’s API to download the last in-sync state and re-set it as the latest state, resolving the issue. (Note this means @jaronoff97 has some extra work to do when finalizing [#274][issue-274].)
+
+Both of these issues from poor coordination and lack of a clear process around infrastructure management, and definitely need more work to come up with long-term solutions. (See [action items](#action-items) at the end of this report.)
+
+
+## Timeline
+
+*(All times US/Pacific, 24-hour clock.)*
+
+### 2021-06-29 18:30
+
+@Mr0grog completes hotfix [#275][issue-275], but discovers actions are not running and all our GitHub checks are failing. He starts trying to figure out what is going wrong and whether this is specific to our repo or a GitHub-wide problem.
+
+
+### 2021-06-29 18:36
+
+Posted a request for help in #team-ops-and-tools in case someone is with better access to the GitHub org account can figure out the issue more clearly. (@Mr0grog does not anticipate this will be very fruitful since it’s after hours, and there’s no clear ops on-call/escalation plan as far as he knows.)
+
+
+### 2021-06-29 19:04
+
+Realized this is a billing issue. Error messages about payment failures are shown on below the fold on the workflow summary screen (not on individual job logs). @Mr0grog updates thread in #team-ops-and-tools with info and a more specific ask about payment.
+
+
+### 2021-06-29 20:41
+
+@Mr0grog gave up on resolving the billing issue and pinged @jaronoff97 for help reviewing the hotfix since automated checks aren’t running.
+
+
+### 2021-06-29 20:43
+
+@jaronoff97 reviews the PR and @Mr0grog merges and starts building & uploading Docker images manually.
+
+
+### 2021-06-29 21:14
+
+@Mr0grog rebuilds and uploads on remote infrastructure. His home internet connection is too slow to upload the build Docker images with any reasonable speed.
```

Comment:
Worth adding a little detail here about the remote infrastructure you used?

Reply:
How’s 26ecfe6def0e61ba3a42365a88edbe1419c0eb84?